src/txorphanage.cpp

```diff
@@ -106,27 +106,27 @@ unsigned int TxOrphanage::LimitOrphans(unsigned int max_orphans)
 {
     AssertLockHeld(g_cs_orphans);
 
-    unsigned int nEvicted = 0;
+    unsigned int nExpired = 0;
     static int64_t nNextSweep;
     int64_t nNow = GetTime();
     if (nNextSweep <= nNow) {
         // Sweep out expired orphan pool entries:
-        int nErased = 0;
         int64_t nMinExpTime = nNow + ORPHAN_TX_EXPIRE_TIME - ORPHAN_TX_EXPIRE_INTERVAL;
         std::map<uint256, OrphanTx>::iterator iter = m_orphans.begin();
         while (iter != m_orphans.end())
         {
             std::map<uint256, OrphanTx>::iterator maybeErase = iter++;
             if (maybeErase->second.nTimeExpire <= nNow) {
-                nErased += EraseTx(maybeErase->second.tx->GetHash());
+                nExpired += EraseTx(maybeErase->second.tx->GetHash());
             } else {
                 nMinExpTime = std::min(maybeErase->second.nTimeExpire, nMinExpTime);
             }
         }
         // Sweep again 5 minutes after the next entry that expires in order to batch the linear scan.
         nNextSweep = nMinExpTime + ORPHAN_TX_EXPIRE_INTERVAL;
-        if (nErased > 0) LogPrint(BCLog::MEMPOOL, "Erased %d orphan tx due to expiration\n", nErased);
+        if (nExpired > 0) LogPrint(BCLog::MEMPOOL, "Erased %d orphan tx due to expiration\n", nExpired);
     }
+    unsigned int nEvicted = nExpired;
```

Comment:
You are now logging expired txs as evicted, see the caller:


```cpp
                // DoS prevention: do not allow m_orphanage to grow unbounded (see CVE-2012-3789)
                unsigned int nMaxOrphanTx = (unsigned int)std::max((int64_t)0, gArgs.GetIntArg("-maxorphantx", DEFAULT_MAX_ORPHAN_TRANSACTIONS));
                unsigned int nEvicted = m_orphanage.LimitOrphans(nMaxOrphanTx);
                if (nEvicted > 0) {
                    LogPrint(BCLog::MEMPOOL, "orphanage overflow, removed %u tx\n", nEvicted);
                }

Reply:
I didn't notice `LimitOrphans` is called here. So it meant to return only the number of evicted tx, though the number is only used for logging.